### DISCUSSION TITLE
Running Trilium on distributed SQLite

### DISCUSSION THREAD

**Discussion by losfair** [TriliumNext/Trilium#3029] [Ideas] [answered]
Hello,

I'm a long time user of Trilium, and first off I want to thank you for making this!

It seems that there are various issues and discussions related to using a networked database with Trilium, like https://github.com/zadam/trilium/issues/174, https://github.com/zadam/trilium/issues/441 and https://github.com/zadam/trilium/issues/457. These proposals weren't accepted because Trilium calls into SQLite directly without an abstraction layer, and I totally understand that. However, personally I have most of my data centrally stored on networked databases like MySQL and [FoundationDB](https://github.com/apple/foundationdb) so that I can apply unified backup strategies on those systems. Previously I had a script that periodically dumps Trilium's database and pushes the dump to S3, but I miss features like PITR backups and synchronous replication in larger DBMSes.

I made [mvsqlite](https://github.com/losfair/mvsqlite) a while ago, which is a drop-in VFS plugin for SQLite that stores multi-versioned data on FoundationDB. I tried running Trilium on it for a while, and with a few small patches it works almost flawlessly. There are still minor differences with pure-upstream SQLite though, since mvsqlite is a MVCC system and existing app logic may not like it very much.

I'm thinking that it would be nice if we can work together to improve the Trilium + mvsqlite combination! I've got a few issue reports and patches, and if you are good with it I'll start sending them :)

**Comment by zadam** [comment, marked as answer]
> I've got a few issue reports and patches, and if you are good with it I'll start sending them :)

Hi, in general I don't plan to support such setups "officially". But if the needed fixes are trivial and don't complicate other stuff, then I'm ok with that. I guess it would be the best if you post some PR to have something specific to talk about.
